### PR TITLE
Support additional FHIR 2.x versions

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions/fhir/deployment/src/main/java/org/apache/camel/quarkus/component/fhir/deployment/dstu2/FhirDstu2Processor.java [hunk 1]
```diff
@@ -42,201 +42,46 @@
...
-                VisionEyesEnum.class.getCanonicalName(),
-                XPathUsageTypeEnum.class.getCanonicalName()
-        };
-    }
+        String[] dstu2Enums = combinedIndex.getIndex()
+                .getKnownClasses()
+                .stream()
+                .map(classInfo -> classInfo.toString())
+                .filter(className -> className.startsWith("ca.uhn.fhir.model.dstu2.valueset"))
+                .toArray(String[]::new);
```
Comment (zbendhiba) [inline extensions/fhir/deployment/src/main/java/org/apache/camel/quarkus/component/fhir/deployment/dstu2/FhirDstu2Processor.java:83]: Nice

File: docs/modules/ROOT/pages/reference/extensions/fhir.adoc [hunk 2]
```diff
@@ -62,7 +62,19 @@ By default, only FHIR versions `R4` & `DSTU3` are enabled in native mode, since
 
 Enable FHIR DSTU2 Specs in native mode.
 | `boolean`
-| `true`
+| `false`
```
Comment (zbendhiba) [inline docs/modules/ROOT/pages/reference/extensions/fhir.adoc:65]: Do we need some migration guide for this one ?

Comment (jamesnetherton) [reply docs/modules/ROOT/pages/reference/extensions/fhir.adoc:65]: Yeah, I can add some notes about the change to the config defaults.

**Review by zbendhiba — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/extensions/fhir.adoc (hunk 2 above)
Comment (jamesnetherton) [reply docs/modules/ROOT/pages/reference/extensions/fhir.adoc:65]: I pushed a commit with a migration guide for 2.8.0.

Comment (zbendhiba) [reply docs/modules/ROOT/pages/reference/extensions/fhir.adoc:65]: Awesome

**Pull request merged**